src/cpu/x64/matmul/brgemm_matmul_utils.cpp

```diff
@@ -182,14 +182,17 @@ status_t check_isa_with_datatype(
         const cpu_isa_t isa, const brgemm_matmul_conf_utils_t &bm_conf_utils) {
     const bool ok
             = IMPLICATION(bm_conf_utils.is_f32(),
-                      one_of(isa, avx512_core, avx2) || bm_conf_utils.is_bf32())
+                      one_of(isa, avx512_core, avx2) || bm_conf_utils.is_bf32()
+                              || bm_conf_utils.is_tf32())
             && IMPLICATION(bm_conf_utils.is_int8(),
-                    is_superset(isa, avx512_core)
+                    is_superset(isa, avx512_core_amx)
```

Comment:
I thought all AMX-based ISA are superset of AVX512_CORE. Which one doesn't fit the description considering there's a check for `avx2_vnni` superset as well?..